src/hal/src/acceleration_structure.rs

```diff
@@ -0,0 +1,224 @@
+//! TODO docs
+
+// TODO remove and add docs
+#![allow(missing_docs)]
+
+use std::ops::Range;
+
+use crate::{buffer::Offset, format::Format, Backend, IndexType};
+
+/// Denotes the type of acceleration structure.
+#[derive(Debug)]
+pub enum AccelerationStructureType {
+    ///
+    TopLevel,
+    ///
+    BottomLevel,
+    // TODO vulkan supports "generic level" (where the concrete build type is specified), but discourages its use for applications "written directly for Vulkan" since it "could affect capabilities or performance in the future" (https://www.khronos.org/blog/vulkan-ray-tracing-final-specification-release). Perhaps this is to better support `vkd3d-proton`, but we probably don't want it exposed in gfx?
```

Comment:
Technically, if there is a possibility of running vkd3d over gfx-portability, then we should try to make it possible.
For example, people could use it to run windows games on macOS.
In this case, we can make a call to not support other levels at least for now, just leaving a comment about this omission.

Reply:
I'll add this back. I wonder if there's a nicer way to handle `Generic`, as it's allowed in the "get sizes" and "create" functions, but not during the "build" commands.